src/systems/labor/build_labor.cairo

```diff
@@ -0,0 +1,492 @@
+#[system]
+mod BuildLabor {
+    use traits::Into;
+    use traits::TryInto;
+    use array::ArrayTrait;
+    use box::BoxTrait;
+
+    use eternum::components::config::WorldConfig;
+    use eternum::components::owner::Owner;
+    use eternum::components::realm::Realm;
+    use eternum::components::realm::RealmTrait;
+    use eternum::components::resources::Resource;
+    use eternum::components::labor::Labor;
+    use eternum::components::labor::LaborTrait;
+    use eternum::components::config::LaborConf;
+    use eternum::components::config::LaborCR;
+    use eternum::components::config::LaborCV;
+    use starknet::ContractAddress;
+    use eternum::constants::WORLD_CONFIG_ID;
+    use eternum::constants::LABOR_CONFIG_ID;
+    use eternum::constants::ResourceIds;
+    use eternum::utils::unpack::unpack_resource_ids;
+
+    use debug::PrintTrait;
+
+    #[external]
+    fn execute(realm_id: felt252, resource_id: u8, labor_units: u128, multiplier: u128) {
+        // assert owner of realm
+        let player_id: ContractAddress = starknet::get_caller_address();
```

Comment:
This var is immediately shadowed without being used.

FWIW I think `get_caller_address` is what we want, not `tx_info.acconut_contract_address`, but I could be wrong.

Reply:
thought i deleted this, sorry
basically get_caller_address will return the world address in dojo. so we need account_contract_address to get the contract calling the world. here's the related discussion : https://discord.com/channels/1062934010722005042/1062934060898459678/1103662308405805188

Reply:
Thanks for pointing me to that conversation 🙏 Prompted by this review I asked a similar question in the cairo-1 discord, wasn't sure about the diff between those two. You're absolutely right, `account_contract_address` is the one we need 👍